### PR TITLE
(WIP) Search Bar

### PR DESCRIPTION
**Pull request by Nick-Gabe**
This Pull Request is not ready yet.

I'm posting it because there is a known bug I didn't found out how to solve.

Basically, every time you type something on the search bar, it searches and show tweets containing what you searched.
image
But when you firstly enters the page, it doesn't show any tweets at all.

Previously, the code who did that job is this:
https://github.com/Nick-Gabe/ccsseraphini/blob/16fd2032e3876771cff32b2bdadadd53e1327be9/apps/web/src/components/tweet/Timeline.tsx#L114
```js
{/* {renderTweets()} */}
```
But if I enable that line and search something, the search will also generate its own "tweet timeline" and then we have 2 divs with the same content.
I was planning to make so when the page loads, it would search a blank space and show the default tweets, but apparently even with `onLoad` it doesn't trigger, only when I type something on it.

Any help would be appreciated, thanks.

### CODE REVIEW

File: apps/web/src/components/tweet/SearchLogo.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+               C3.902,25.675,3.904,15.902,9.907,9.905c6.003-6.002,15.77-6.002,21.771-0.003c5.999,6,5.997,15.762,0,21.774
+               C25.676,37.66,15.91,37.682,9.905,31.678z"
+          />
+          <path
+            d="M14.18,22.464c-0.441-1.812-2.257-4.326-3.785-3.525c-1.211,0.618-0.87,3.452-0.299,5.128
+               c2.552,7.621,11.833,9.232,12.798,8.268C23.843,31.387,15.928,29.635,14.18,22.464z"
+          />
+        </g>
+      </g>
+    </svg>
```
Comment (EternalQuasar0206) [inline apps/web/src/components/tweet/SearchLogo.tsx:37]: Can't you use the SVG externally for code decoupling/possible reuses in the future?

File: apps/web/src/components/tweet/Timeline.tsx [hunk 2]
```diff
@@ -105,8 +110,8 @@ export const Timeline = ({
       </Text>
 
       {refresh ? <Spinner position="absolute" top="15%" right="50%" /> : null}
-
-      {renderTweets()}
+      <TweetSearch />
+      {/* {renderTweets()} */}
```
Comment (EternalQuasar0206) [inline apps/web/src/components/tweet/Timeline.tsx:114]: code smell? It is not a good idea to keep the unused code commented, you can recover the commit history anytime

File: apps/web/src/components/tweet/Timeline.tsx [hunk 3]
```diff
@@ -105,8 +110,8 @@ export const Timeline = ({
       </Text>
 
       {refresh ? <Spinner position="absolute" top="15%" right="50%" /> : null}
```
Comment (EternalQuasar0206) [inline apps/web/src/components/tweet/Timeline.tsx:112]: `{refresh && <Spinner position="absolute" top="15%" right="50%" />}` would be cooler

**Comment by victoriaquasar** [comment]
Pq fechou
Tá de memes comigo?

**Comment by Nick-Gabe** [comment]
Por agora não conseguiria continuar o pr, mas se quiser pode reabrir e outra pessoa talvez finalizar então

**Pull request closed without merging**